### PR TITLE
Moved HtmlAttributeIRNode and children from renderer to writer

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@NTaylorMullen @rynowak 

Added an internal abstract `HtmlAttributeWriter`. Will add tests once design is approved.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.CodeGeneration
+{
+    internal abstract class HtmlAttributeWriter
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs]: Any reason the `WriteHtmlAttribute` shouldn't also be on here?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs]: This is an internal class. Don't want to expose a way for people to override the HtmlAttribute code generation?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs]: Actually I think we could rename this to `HtmlAttributeValueWriter`.

**Comment by ajaybhargavb** [comment]
:up: 📅 
- Removed RenderingConventions
- Added Redirect writers
- Updated an integration test to verify redirect scenario for Html attributes.
- Cleanup

File: src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/HtmlAttributeWriter.cs]: All of this should be on BasicWriter

**Comment by ajaybhargavb** [comment]
🆙 📅 No longer in design.

- Addressed feedback
- Added tests

File: test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/TagHelperHtmlAttributeRuntimeBasicWriterTest.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+@"AddHtmlAttributeValue("" "", 27, new Microsoft.AspNetCore.Mvc.Razor.HelperResult(async(__razor_attribute_value_writer) => {
+    Render Children
+}
+), 28, 13, false);
+",
+                csharp,
+                ignoreLineEndingDifferences: true);
+        }
+
+        private static CSharpRenderingContext GetCSharpRenderingContext(BasicWriter writer)
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/TagHelperHtmlAttributeRuntimeBasicWriterTest.cs]: Any possibility to make a test base class to hold all these infrastructure bits? It looks like they're duplicated quite a bit

Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/TagHelperHtmlAttributeRuntimeBasicWriterTest.cs]: I don't think it's going to add a lot of value. `GetCSharpRenderingContext` does different things for BasicWriter and TagHelperWriter. Only `Lower` can be shared. Is it worth it? I don't know........

Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/TagHelperHtmlAttributeRuntimeBasicWriterTest.cs]: Ok fair enough.

**Pull request merged**